content/post/051-new-ipfs-release-process.md

```diff
@@ -0,0 +1,229 @@
+---
+date: 2019-07-18
+url: 2019-07-18-ipfs-release-process
+tags: IPFS, go-ipfs, release process, 
+title: Improving the IPFS Release Process
+author: Steven Allen, Alan Shaw, David Dias, Molly Mackinlay
+---
+
+go-ipfs is introducing a new release cycle and process to ensure reliable releases.
+
+## Background
+
+When reflecting on Q2, we noticed an unusual number of critical regressions
+(since fixed) introduced in the last three go-ipfs releases:
+
+* **go-ipfs 0.4.21** had two performance regressions in bitswap:
+  1. A throughput regression that should have been caught by regression testing
+    (now tested) but almost certainly would have been noticed by downstream
+    users given a longer release-testing process.
+  2. A CPU utilization regression that only shows up with >10000 peers. Something
+    that only really shows up in certain production systems under heavy load.
+* **go-ipfs 0.4.20** had a regression where adding multiple independent files in the same
+  add command didn't work ([#6254](https://github.com/ipfs/go-ipfs/pull/6255)).
+  A regression test has since been added, but this _also_ would have been caught
+  with better cross-application testing.
+* **go-ipfs 0.4.19** had multiple regressions (all fixed in 0.4.20):
+  1. A regression in the docker container (introduced by
+    [#6040](https://github.com/ipfs/go-ipfs/pull/6040)) that would have been
+    caught by testing the go-ipfs docker image in more production environments.
+  2. A CPU utilization regression in bitswap only seen under very high load. This
+    would have been caught by testing under production loads.
+  3. Panics in the DHT and QUIC modules that only show up under heavy load.
```

Comment:
Do you think this should be chronological? IMHO it feels wrong to read this from newest to oldest.

Reply:
[`0.4.19` was built 1st March](https://dist.ipfs.io/go-ipfs/v0.4.19/build-info) so is technically not Q2. Perhaps we should say we were reflecting on Q1 **and** Q2?

That would actually help because it would mean our proposal for 6 week release cycle would actually be faster than if these 3 were all released in one quarter! 😂

Reply:
I'll change it to "reflecting on this year so far".